streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -23,11 +23,13 @@
 import java.util.Map;
 
 /**
- * The window specifications used for joins.
+ * The window specifications used for joins. TODO extend
  */
 public class JoinWindows extends Windows<TumblingWindow> {
 
+    /** Maximum time difference for tuples that are before the join tuple. */
```

Comment:
Actually, having thought more about it, there's nothing in Kafka Streams that would enforce/require milliseconds or millseconds-since-the-epoch semantics.  (Note that, for the embedded timestamps introduced in KIP-32, the contract of Kafka is to have these timestamps represent milliseconds-since-the-epoch).

For example, a user could embed minute-based timestamps into her Kafka messages, then define a custom timestamp extractor that would return these minute-based timestamps as-is (i.e. no need to multiply by `60 * 1000` to convert to milliseconds), and any subsequent configure window sizes etc. based on numbers representing minutes -- and it would work just fine.  Correct, @guozhangwang ?